fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -521,13 +524,61 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
      * thread, early in the app startup sequence.
      */
     private fun beginSetupMegazord() {
-        // Note: Megazord.init() must be called as soon as possible ...
-        Megazord.init()
+        try {
+            // Note: Megazord.init() must be called as soon as possible ...
+            Megazord.init()
+
+            initializeRustErrors(components.analytics.crashReporter)
+            // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
 
-        initializeRustErrors(components.analytics.crashReporter)
-        // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
+            RustLog.enable()
+        } catch (e: UnsatisfiedLinkError) {
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                reportUnsatisfiedLinkErrorBreadcrumbs()
+            } catch (e: Throwable) {
+                // This shouldn't happen, but if it does it's better to ignore the exception from
+                // the breadcrumb code and rethrow the initial exception.
+            }
+            throw e
+        }
+    }
 
-        RustLog.enable()
+    private fun reportUnsatisfiedLinkErrorBreadcrumbs() {
+        val breadcrumbs = mutableListOf<String>()
+        val apkPath = applicationContext.getApplicationInfo().sourceDir
+        breadcrumbs.add("APK: $apkPath")
+        val apkDir = File(apkPath).getParentFile()
+        val installSourcePackage = if (SDK_INT >= Build.VERSION_CODES.R) {
+            packageManager.getInstallSourceInfo(packageName).installingPackageName
+        } else {
+            @Suppress("DEPRECATION")
+            packageManager.getInstallerPackageName(packageName)
+        }
+        breadcrumbs.add("Installer package name: $installSourcePackage")
+
+        val apkDirFileList = apkDir?.let {
+            it.walk()
+                .filter { it != apkDir }
+                .map { it.relativeTo(apkDir).toString() }
+                .joinToString(", ")
+        }
+        breadcrumbs.add("Files inside dir: $apkDirFileList")
+
+        val apkFileList = ZipInputStream(FileInputStream(apkPath)).use {
+            generateSequence { it.nextEntry }.joinToString(", ")
+        }
+        breadcrumbs.add("Files inside APK: $apkFileList")
+
+        for (breadcrumb in breadcrumbs) {
+            components.analytics.crashReporter.recordCrashBreadcrumb(
+                Breadcrumb(
+                    category = "Startup",
+                    message = breadcrumb,
+                    level = Breadcrumb.Level.INFO,
+                ),
+            )
```

Comment:
These should show up on Sentry events, but not Soccoro, is there a way to get this data inside those reports too?

Reply:
Unfortunately, this will require some work to support.  Currently breadcrumbs is only supported for Socorro reports for caught exception.  For uncaught exceptions, work is needed to pass these breadcrumbs to the crash reporting service. (Or persist on disk and cleared after reporting)

Reply:
This is the [issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1792898) tracking this work.